typing/typeclass.ml

```diff
@@ -48,20 +48,20 @@ type class_type_info = {
 }
 
 type 'a class_res = {
```

Comment:
I was thinking if we could come up with a more expressive name for this record type.

Reply:
The name isn't actually used, so I wouldn't worry about it too much! I'm not familiar with the type checker - isn't this record ultimately the result of typing an `object` expression (as opposed to `Typing.class_type_declaration` which is for the type of the `class` itself). In which case might `'a object_expression` or `'a object_declaration` work?

Reply:
I'm not sure either (maybe @lpw25 would know), but from looking at the way this data is used in the module I would still call this `class_decl` or `class_declaration`. The producer of this record is `final decl`, which I understand as returning the "final version of the declaration", called by the `type_classes` function which deals with a mutually-recursive block of class declarations (according to its name and the name of its callers).

Reply:
> isn't this record ultimately the result of typing an object expression 

Nope, this is for the full class or class type. (By the way "object expression" is normally called either "class signature"/"class structure" to distinguish it from "object literal" -- which is when a class structure is used directly as an expression.)